packages/nimbus/src/components/money-input/money-input.dev.mdx

```diff
@@ -0,0 +1,427 @@
+
+```jsx-live-dev
+const App = () => {
+  const [value, setValue] = useState<MoneyInputValue>({
+    amount: "123.45",
+    currencyCode: "USD"
+  });
+
+  return (
+    <Stack direction="column" gap="400">
+      <MoneyInput
+        size="sm"
+        value={value}
+        currencies={["USD", "EUR"]}
+        onValueChange={setValue}
+      />
+      <MoneyInput
+        size="md"
+        value={value}
+        currencies={["USD", "EUR"]}
+        onValueChange={setValue}
+      />
+    </Stack>
+  );
+}
+```
+
+### Currency configuration
+
+The `currencies` prop controls the available options in the dropdown. If no currencies are provided (or an empty array is passed), the component renders a static label for the currency code instead of a dropdown.
+
+```jsx-live-dev
+const App = () => {
+  const [value1, setValue1] = useState<MoneyInputValue>({
+    amount: "100.00",
+    currencyCode: "USD"
+  });
+
+  const [value2, setValue2] = useState<MoneyInputValue>({
+    amount: "50.00",
+    currencyCode: "EUR"
+  });
+
+  return (
+    <Stack direction="column" gap="400">
+      <Text fontWeight="bold">Multiple currencies (Dropdown)</Text>
+      <MoneyInput
+        value={value1}
+        currencies={["USD", "EUR", "GBP", "JPY"]}
+        onValueChange={setValue1}
+      />
+
+      <Text fontWeight="bold">No currencies list (Static Label)</Text>
+      <MoneyInput
+        value={value2}
+        currencies={[]}
+        onValueChange={setValue2}
+      />
+    </Stack>
+  );
+}
+```
+
+### High precision support
+
+The `MoneyInput` automatically detects high precision values (more decimal places than standard for the currency) and displays a "High Precision" badge. This feature can be toggled with `hasHighPrecisionBadge`.
+
+```jsx-live-dev
+const App = () => {
+  const [value, setValue] = useState<MoneyInputValue>({
+    amount: "123.4567",
+    currencyCode: "USD"
+  });
+
+  return (
+    <Stack direction="column" gap="400">
+      <Text fontSize="sm">
+        Try entering more than 2 decimal places for USD (e.g., 123.4567)
+      </Text>
+      <MoneyInput
+        value={value}
+        currencies={["USD", "EUR"]}
+        onValueChange={setValue}
+        hasHighPrecisionBadge={true}
+      />
+    </Stack>
+  );
+}
+```
+
+### State variations
+
+The component supports standard interactive states including disabled, read-only, and invalid.
+
+```jsx-live-dev
+const App = () => {
+  const value = { amount: "123.45", currencyCode: "USD" };
+
+  return (
+    <Stack direction="column" gap="400">
+      <MoneyInput
+        label="Disabled"
+        value={value}
+        currencies={["USD"]}
+        isDisabled
+      />
+      <MoneyInput
+        label="Read-only"
+        value={value}
+        currencies={["USD"]}
+        isReadOnly
+      />
+      <MoneyInput
+        label="Invalid"
+        value={value}
+        currencies={["USD"]}
+        isInvalid
+      />
+    </Stack>
+  );
+}
+```
+
+### Modern event handlers
```

Comment:
I'm not sure we need to use the term "Modern" here. I think we can just use "Event handlers". What do you think.